### PR TITLE
chore: use collapsible tab views inside profile tab

### PR DESCRIPTION
**Pull request by MounirDhahri**
### Description

This PR migrates all tabs inside profile tab to use the new tabs from palette-mobile. This PR makes the following changes:
- [x] Extract sticky content into separate components instead of injecting it using `setJSX`
- [x] Use `Tabs.ScrollView` inside Profile Tabs
- [x] Fix some nested ScrollView warning inside my collection insights
- [x]

Recommended next steps:
- [ ] Remove old sticky tabs code
- [x] Add visual clues logic into palette tabs [**needed for features parity with what we currently have**]
- [ ] Add nice placeholder to insights tab (**not blocking**)

**Videos**
- Tabs 

https://github.com/artsy/eigen/assets/11945712/19ccf7ee-b75b-4c0a-8891-b0df69394add

___
- Banners

https://github.com/artsy/eigen/assets/11945712/23123532-ab92-4cdc-b121-65820858bbc2

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- use collapsible tab views inside profile tab - mounir

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/app/Scenes/MyCollection/MyCollectionArtworks.tsx [hunk 1]
```diff
@@ -64,31 +50,10 @@ export const MyCollectionArtworks: React.FC<MyCollectionArtworksProps> = ({
...
-
-      setShowSearchBar(true)
-    }
-  }
-
   return (
     <Flex minHeight={minHeight} px={2}>
       <Flex mb={1}>
-        {!!showSearchBar && (
+        {artworks.length > 4 && (
```
Comment (MounirDhahri) [inline src/app/Scenes/MyCollection/MyCollectionArtworks.tsx:56]: This logic was broken - the requirement was to show this only when the user has more than 4 artworks - the fancy logic here is kind of broken so better remove it

File: src/app/Scenes/MyCollection/Screens/Insights/MedianAuctionPriceRail.tsx [hunk 2]
```diff
@@ -46,6 +46,8 @@ export const MedianAuctionPriceRail: React.FC<MedianAuctionPriceRailProps> = (pr
       <FlatList
         data={groupedArtworks}
         listKey="median-sale-prices"
+        scrollEnabled={false}
+        nestedScrollEnabled={false}
```
Comment (MounirDhahri) [inline src/app/Scenes/MyCollection/Screens/Insights/MedianAuctionPriceRail.tsx:50]: same thing here

File: src/app/Scenes/MyCollection/Screens/Insights/AuctionResultsForArtistsYouCollectRail.tsx [hunk 3]
```diff
@@ -45,6 +45,8 @@ export const AuctionResultsForArtistsYouCollectRail: React.FC<
       <FlatList
         data={auctionResultsData}
         listKey="artist-auction-results"
+        nestedScrollEnabled={false}
```
[the remark this answers is not in the document]
Comment (damassi) [reply src/app/Scenes/MyCollection/Screens/Insights/AuctionResultsForArtistsYouCollectRail.tsx:48]: Ah, thanks for the tip, was not aware of this. It prevents that virtualization issue previously mentioned, yeah?

**Comment by damassi** [comment]
Amazing, thanks so much @MounirDhahri 👍 

Will add visual queue support and then open a PR against this PR.

File: src/app/Scenes/MyProfile/MyProfileHeaderMyCollectionAndSavedWorks.tsx [hunk 4]
```diff
@@ -36,43 +35,29 @@ interface MyProfileTabProps {
 export const MyProfileHeaderMyCollectionAndSavedWorks: React.FC<Props> = ({ initialTab }) => {
   const isArtworkListsEnabled = useCheckIfArtworkListsEnabled()
   const viewKind = MyCollectionTabsStore.useStoreState((state) => state.viewKind)
-  const tabs: TabProps[] = compact([
```
Comment (MounirDhahri) [inline src/app/Scenes/MyProfile/MyProfileHeaderMyCollectionAndSavedWorks.tsx:39]: this is the code related to the visual clues. These clues are used as a way to notify users about something new available for them. 
Screenshot 2023-06-06 at 17 25 27

**Review by damassi — APPROVED** [review]
Indicators stuff is in so going to merge this!

**Review by dariakoko — COMMENTED** [review]
Thank you Mounir! Looks good
I only have two comments

- the way the content of the tabs is loading looks inconsistent - placeholders on MyCollection and Saves tabs vs loading indicator on the Insights tab

- when adding the artwork by Amoako Boafo there should be a visual clue on the Insights tab, shouldn't it? I tried adding the same artwork with the app version 8.13.0 build 2023.05.25.07 and I see the visual clue

**Comment by MounirDhahri** [comment]
@dariakoko thanks for QA'ng this in App. Both your comments are valid and it makes sense to address them before release. I will create tickets for both issues, add them to the sprint and work on them today.

**Pull request merged**